### PR TITLE
Cleaned up and updated Weather Sample

### PR DESCRIPTION
**Pull request by BenBtg**

### CODE REVIEW

File: weather/xamarin/Weather/Services/MainThreadAsyncService.cs [hunk 1]
```diff
@@ -11,7 +10,7 @@ public class MainThreadAsyncService : IMainThreadAsyncService
         public Task<T> RunOnMainThreadAsync<T>(Func<Task<T>> func)
         {
             var tcs = new TaskCompletionSource<T>();
-            MainThread.BeginInvokeOnMainThread(async () =>
+            MainThread.Run(async () =>
```
Comment (DeanFaizal) [inline weather/xamarin/Weather/Services/MainThreadAsyncService.cs]: Should this be RunAsync instead?

File: weather/xamarin/Weather/Weather.csproj [hunk 2]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <AssemblyName>Weather.Forms</AssemblyName>
+    <UserSecretsId>8834abd2-d94c-4850-8f31-75d07e4518ff</UserSecretsId>
```
Comment (DeanFaizal) [inline weather/xamarin/Weather/Weather.csproj]: Is this intentional?

Comment (Sweekriti91) [reply weather/xamarin/Weather/Weather.csproj]: Good question. What is this for?

**Comment by Sweekriti91** [comment]
Any problems if I leave this open till I get the pipelines running again? I want to make sure we aren't breaking stuff 😂 I'll approve for merge once build sanity check passes.

File: weather/xamarin/Weather/Services/MainThreadAsyncService.cs (hunk 1 above)
Comment (BenBtg) [reply weather/xamarin/Weather/Services/MainThreadAsyncService.cs]: Yes it should!

File: weather/xamarin/Weather.Android/Weather.Android.csproj [hunk 3]
```diff
@@ -153,5 +149,5 @@
   <ItemGroup>
     <ProguardConfiguration Include="proguard.cfg" />
   </ItemGroup>
- <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.CSharp.targets" />
-</Project>
+  <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.CSharp.targets" />
+</Project>
```
[the remark this answers is not in the document]
Comment (BenBtg) [reply weather/xamarin/Weather.Android/Weather.Android.csproj]: Must be white space.

File: weather/xamarin/Weather/Weather.csproj (hunk 2 above)
Comment (BenBtg) [reply weather/xamarin/Weather/Weather.csproj]: I have no idea where that came from!
Think VS2019 generated it when I checking out the menu option.

**Comment by DeanFaizal** [comment]
Good stuff!

**Pull request merged**